backend/server.js

```diff
@@ -15,13 +37,66 @@ const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+const authenticateUser = async (req, res, next) => {
+try {  
+  const user = await User.findOne({ 
+    accessToken: req.header('Authorization') 
+  })
+  if (user) {
+    req.user = user
+    next()
+  } else {
+    res.status(401).json({ loggedOut: true, message: 'Please try login again' })
+  }
+} catch (err) {
+  res
+    .status(403)
+    .json({ message: 'access token missing or wrong', errors: err.errors })
+}
 
 // Start defining your routes here
 app.get('/', (req, res) => {
   res.send('Hello world')
+ });
+
+ app.post('/register', async (req, res) => {
+  try {
+    const { name, email, password} = req.body;
+    const user = new User({ name, email, password: bcrypt.hashSync(password) });
+    const saved = await user.save();
+    res.status(201).json(saved);
+  } catch (err) {
+    res.status(400).json({ message: 'could not save user', error: err.errors })
+  }  
+});
+
+ app.get('/users/:id', authenticateUser)
+ app.get('/users/:id', (req, res) => {
+   try {
+     res.status(201).json(req.user)
+   } catch (err) {
+     res.status(400).json({message: 'could not save user', errors: err.message})
+   }
+ })
+
+// login user
+app.post('/sessions', async (req, res) => {
+  try {
+    const { name, password } = req.body
```

Comment:
Another issue is here:  You are expecting name from the `req.body`, but on 
https://github.com/Technigo/project-auth/pull/12/files#diff-3c712706d81b44f9660bc1b5cf8b1fa9R17

You are sending `{email, password}` from your frontend so mongo will not find the correct user when looking for a matching email with the name value

Reply:
Fixed :)